pkg/resolution/common/interface.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2023 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+
+	pipelinev1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ResolverName is the type used for a resolver's name and is mostly
+// used to ensure the function signatures that accept it are clear on the
+// purpose for the given string.
+type ResolverName string
```

Comment:
move from: https://github.com/tektoncd/pipeline/blob/40763d8a3ee63286e265b8fb485af4354bbe6463/pkg/resolution/resource/resource.go#L26-L63

avoid cycle import. 😁

The test package references these interfaces.
https://github.com/tektoncd/pipeline/blob/40763d8a3ee63286e265b8fb485af4354bbe6463/test/resolution.go#L12-L12

Reply:
Although moving the interface to write unit tests is not reasonable, since the methods in the test package are referenced in multiple places, I can only choose the way to make the minimum changes.

And I also need to use existing methods in the test package, I don't want to duplicate them here.